amads/core/basics.py

```diff
@@ -827,6 +827,83 @@ def __init__(self, delta=0, duration=0, content=None, time_map=None):
         super().__init__(delta, duration, content)
         self.time_map = time_map if time_map else TimeMap()
 
+    @classmethod
+    def from_melody(cls, pitches, deltas=None, durations=None):
+        """Create a Score from lists of pitches, deltas, and durations.
+
+        Parameters
+        ----------
+        pitches : list of int
+            MIDI note numbers for each note
+        deltas : list of float, optional
+            Start times in quarters relative to the score's start. If not provided,
+            defaults to [0, 1, 2, ...].
+        durations : list of float, optional
+            Durations in quarters for each note. If not provided, defaults to
+            1.0 for each note.
+
+        Returns
+        -------
+        Score
+            A new Score object containing the melody in a single part
+
+        Raises
+        ------
+        ValueError
+            If input lists have different lengths or are empty
```

Comment:
What if durations and deltas are in conflict?

Reply:
Mmm, good point. I agree, how about I replace deltas with iois? I think iois would probably be easier to use in most cases.

Reply:
I agree, and maybe raise an error if deltas and iois are both provided? You could allow them to both be present, but with floating point rounding error, it's not clear what it means for these to be consistent, e.g. what if they differ by 1 microsecond?

Reply:
I agree, we can still support deltas as a possible argument, and I agree that it makes sense to throw an error if both are provided.

Reply:
I've now implemented this, including various sensible defaults. See what you think?

Reply:
Looks good, but why not allow for empty note lists creating an empty score?

Reply:
You're absolutely right, I've now implemented support for this. Let me know if you would like me to change anything else, otherwise I'll go ahead and merge.

Reply:
go ahead and merge -- empty scores was the only question/comment I had.

Reply:
Thank you!